tools/asset-sync/assets.ps1

```diff
@@ -0,0 +1,521 @@
+#>
+Function Initialize-AssetsRepo {
+    param(
+        [Parameter(Mandatory=$true)]
+        [PSCustomObject] $Config,
+        [Parameter(Mandatory=$false)]
+        [boolean] $ForceReinitialize = $false
+    )
+    $assetRepo = Resolve-AssetRepo-Location -Config $Config
+    $initialized = Is-AssetsRepo-Initialized -Config $Config
+    $workCompleted = $false
+
+    if ($ForceReinitialize)
+    {
+        Remove-Item -Force -R "$assetRepo/*"
+        $initialized = $false
+    }
+
+    if (!$initialized){
+        try {
+            Push-Location $assetRepo
+        
+            Write-Host "git clone --no-checkout --filter=tree:0 https://github.com/$($Config.AssetsRepo) ."
+            git clone --no-checkout --filter=tree:0 https://github.com/$($Config.AssetsRepo) .
+
+            $targetPath = Resolve-CheckoutPaths -Config $Config
+            $targetBranch = Resolve-TargetBranch -Config $Config
+
+            Write-Host "git sparse-checkout init"
+            git sparse-checkout init
+
+            Write-Host "git sparse-checkout set '/*' '!/*/' $targetPath"
+            git sparse-checkout set '/*' '!/*/' $targetPath
+            
+            Write-Host "git checkout $($Config.SHA)"
+            git checkout $($Config.SHA)
+            
+            if($LASTEXITCODE -gt 0){
+                throw "Unable to clone to directory $assetRepo"
+            }
+            $workCompleted = $true
+        }
+        finally {
+            Pop-Location
+        }
+    }
+
+    return $workCompleted
+}
+
+<#
+.SYNOPSIS
+This function will forcibly reset the repo to a targeted SHA. This is a **destructive** update.
+
+.DESCRIPTION
+#>
+Function Reset-AssetsRepo {
+    param (
+        [Parameter(Mandatory = $true)]
+        $Config
+    )
+    try {
+
+        $assetRepo = Resolve-AssetRepo-Location -Config $Config
+        Push-Location  $assetRepo
+
+        Write-Host "git checkout *"
+        git checkout *
+        Write-Host "git clean -xdf"
+        git clean -xdf
+        Write-Host "git reset --hard (Get-Default-Branch)"
+        git reset --hard (Get-Default-Branch)
+
+        # need to figure out the sparse checkouts if we want to optimize this as much as possible
+        # for prototyping checking out the whole repo is fine
+        if($AssetsRepoSHA){
+            Write-Host "git checkout $AssetsRepoSHA"
+            git checkout $AssetsRepoSHA
+            Write-Host "git pull"
+            git pull
+        }
+    }
+    finally {
+        Pop-Location
+    }
+}
+
+<#
+.SYNOPSIS
+This function's purpose is solely to update a assets.json (both config and on file) with a new recording SHA.
+
+.DESCRIPTION
+Retrieves the location of the target recording.json by looking at a property of the Config object.
+#>
+Function Update-AssetsJson {
+    param(
+        $Config,
+        $NewSHA
+    )
+    
+    $jsonAtRest = Get-Content $Config.AssetsJsonLocation | ConvertFrom-Json
```

Comment:
Similar to above, without `-Raw` this does not return a string but a newline delimited array of strings.

```suggestion
    $jsonAtRest = Get-Content -Raw -Path $Config.AssetsJsonLocation | ConvertFrom-Json
```